### ISSUE TITLE
arXiv '20 | Big Self-Supervised Models are Strong Semi-Supervised Learners

### ISSUE THREAD

**Issue by vycezhong** [vycezhong/read-papers#92]
https://arxiv.org/pdf/2006.10029.pdf

SimCLRv2

**Comment by vycezhong** [comment]
SimCLRv2这篇主要是提出了一个新的范式: pretrain -> finetune -> distill

主要idea是在pretrain和finetune的时候搞一个很大的模型去学习，最后再做一个distill回到小模型，涨点很凶猛

![image](https://user-images.githubusercontent.com/25879526/89263103-39ea8a80-d663-11ea-8a90-bb7f0a8f2917.png)

本文基本也是堆砌实验结论，而且有些结论都没给解释……写的不如上篇 #83 

## SimCLRv2 

### pretrain

几个和原来不同的地方：
1. 用大模型，最大的是ResNet-152(3x + SK)，在1%label上finetune上相比ResNet50提高了大约17个点. 
2. projection head从2层MLP换到了3层，并且最后只抛弃最后两层，保留第一层MLP. 相对提高14%. 
3. 加了MoCo的memory bank，但是只涨了一个点. 因为本身batch size够大了，能提供足够的负样本。

### finetune

就是finetune，如前所述，取head的第一层的输出作为feature,后面再跟一层linear输出到class
![image](https://user-images.githubusercontent.com/25879526/89265585-4d97f000-d667-11ea-9687-4a1aa522b6bd.png)


### distillation

也没啥特别的，学习soft-output

![image](https://user-images.githubusercontent.com/25879526/89265496-2b9e6d80-d667-11ea-82c3-0fdbce4a3d58.png)

也可以加上一定的监督loss, 但这个最后没用
![image](https://user-images.githubusercontent.com/25879526/89265569-47a20f00-d667-11ea-9b56-de7391e74618.png)

## Some tricks

**settings**

我对实验的setting格外敏感。。。这里还是用了square root LR scaling，5%的epochs做warmup，之后cos decay; batch size 4096，训练800个epochs; finetune 1%的label用60个epochs；finetune 10%的label用30个epochs; distillation分两个：self-distillation和big-to-small distillation

然后总结了三个claim 

1. Bigger Models are more label-efficient 
![Screenshot-20200804153351-1366x532](https://user-images.githubusercontent.com/25879526/89265996-ea5a8d80-d667-11ea-9e38-8be5b36cc994.png)

如图，同样改变模型，supervised只涨了4个点不到，而对比学习却涨了17个点。尤其是resnet50在10%的label下取得了68.4%，而resnet152(2x)只需要1%的label就能到70.2%，而Resnet50大小24M,而resnet152(2x)有233M，换句话说，模型大了10倍，所需要的标签是原来十分之一。这就是为什么说大模型更加label efficient。

2. Bigger/Deeper Projection Heads Improve Representation Learning.

简单来说就是发现head越复杂finetune的效果越好，而且存在一个最优的中间层，一般是第一层后面，出来的feature最棒。
 
![Screenshot-20200804153728-1072x700](https://user-images.githubusercontent.com/25879526/89266380-6bb22000-d668-11ea-9904-e16de3918442.png)

没解释为什么，就是经验结论，玄学。

3. Distillation Using Unlabeled Data Improves Semi-Supervised Learning

这个是发现其实模型蒸馏用无监督的数据就够了，如下图

![Screenshot-20200804153948-938x258](https://user-images.githubusercontent.com/25879526/89266632-bf246e00-d668-11ea-9cc9-609caa24c595.png)

Distillation loss就是上文的L^{distill}. 

主要在两个方面，1）大模型可以把学到知识教给小模型 2）self-distillation都能涨点。所以在大模型教小模型之前，大模型会自己先self-distillation。
![image](https://user-images.githubusercontent.com/25879526/89267400-c0a26600-d669-11ea-86a3-b16384399633.png)


### 其他一些发现

1. 在semi-supervised的情况下，Bigger Model只有在 1）用了一些正则方法，如数据增强，label smoothing，2）预训练 的时候才有用，对于supervised learning, 其实没啥用。
![image](https://user-images.githubusercontent.com/25879526/89268020-89808480-d66a-11ea-9490-6351acbff496.png)

2. Linear evaluation的结果与finetune结果是线性相关的，且label越多，斜率越小
![image](https://user-images.githubusercontent.com/25879526/89268087-a1580880-d66a-11ea-8cdb-948f8e2d76e3.png)



## Experiments

当然是吊打了，还吊打了BYOL #95 
![image](https://user-images.githubusercontent.com/25879526/89267121-62758300-d669-11ea-9620-6befdc8124e0.png)

Resnet50在1%label上能到73.9%，比原来SimCLRv1的48.3%高了一大截，我估计主要原因还是因为大模型预训练+蒸馏，这个方法确实work。

___ 

主要贡献还是这个pretrain -> finetune -> distill的范式，很有意思，NLP比较常见，之前一大堆对bert的蒸馏工作，如 #5 

然后是发现前两步需要用大模型，因为label很少，一般认为大模型容易过拟合，但是似乎大模型在这种情况下，不但没有过拟合，反而泛化效果更好。

缺点可能还是800个epochs, 太惊人了，训练一次要多久，要多少钱
